mitodl/ocw-hugo-themes
# more ocw-studio updates

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-hugo-themes/issues/190
Closes https://github.com/mitodl/ocw-hugo-themes/issues/188
Closes https://github.com/mitodl/ocw-hugo-themes/issues/189
Closes https://github.com/mitodl/ocw-hugo-themes/issues/191

#### What's this PR do?
In https://github.com/mitodl/ocw-to-hugo/pull/366, we made a number of adjustments to `ocw-to-hugo` to render course data template properties more in line with the way `ocw-studio` publishes them:

 - `departments` has been removed and replaced with `department_numbers` which is a simple list of the department numbers with a search URL generated by `get_search_url.html`
 - `extra_course_numbers` is now a single string with a comma separated list of the extra course numbers
 - the search URL has been removed from `course_features` objects and it is now a simple list of strings and the key has been renamed to `learning_resource_types`
 - the search URL for `level` is now generated by `get_search_url.html` and it is now just a simple string representing the course level

This PR adjusts the layouts to accommodate these changes and adds some data templates and partials for looking up departments and search URL's.

#### How should this be manually tested?
 - Read the readme if you've never spun up a course site before and make sure you have S3 access set up
 - Make sure you run `yarn install` to make sure the latest `ocw-to-hugo` is installed as a dependency
 - In your `.env` file, make sure `OCW_TO_HUGO_PATH` and `OCW_TO_HUGO_OUTPUT_DIR` are blank and `OCW_TEST_COURSE` is set to a course id (I used `18-06-linear-algebra-spring-2010` but feel free to test others)
 - Spin up the course site with `npm run start:course`
 - Browse the course site and verify that everything shows up as expected in the course info section.  Compare it to the live course on https://ocwnext.odl.mit.edu/.  Click around to the different pages in the course and make sure there are no broken links.  Note that the search links will not work.
